### PR TITLE
Remove twisted-names and twisted-core.

### PR DESCRIPTION
**Pull request by PikkuJose**
Remove twisted-names and twisted-core dependencies as upstream
seems to be moving to a single package distribution.

Signed-off-by: José Pekkarinen <koalinux@gmail.com>

### CODE REVIEW

**Comment by mgorny** [comment]
Thank you for your change. The commits do not conform to our coding standards and it's so trivial it doesn't justify asking you to fix them, so we'll just fix it ourselves while merging.

Given that the package doesn't have tests enabled, did you verify that it indeed works correctly with the current versions of split twisted packages?

**Comment by PikkuJose** [comment]
Hi @mgorny, I'm happy to learn about the non conformed code. If you remove the packages from master I'll adopt them in my overlay because they are dependencies for blink.

Yes, I tested them, I didn't see any broken thing in my blink setup, using that fix, as otherwise portage will prevent me updating properly.

Thanks for the comments!

José.

**Comment by mgorny** [comment]
Ok, if there's actual interest in those packages, it changes things. Would you be interested in maintaining them? Also, are you interested in adding Blink to Gentoo then? I'll elaborate more on what needs to be done when I know how far I should go with that ;-).

**Comment by mgorny** [comment]
Thanks good to hear. So for a start, please look at GLEP 66 (or just a few existing commits in gentoo.git) for how commit messages should look like. Make sure to do separate commits for each package and each logical change.

Other things to note initially:
* please update copyright date on top of each file when touching it for the first time,
* the packages need bumping to newest versions,
* the packages should have tests if possible (read https://wiki.gentoo.org/wiki/Project:Python/Tests),
* please remove python@ from metadata.xml and add yourself + proxy-maint project instead.

**Comment by PikkuJose** [comment]
Aw... forgot the copyright... let me recheck it :)

File: dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild [hunk 1]
```diff
@@ -20,7 +20,11 @@ KEYWORDS="~amd64 ~x86"
...
 RDEPEND="
 	dev-python/greenlet[${PYTHON_USEDEP}]
-	dev-python/twisted-core[${PYTHON_USEDEP}]
+	dev-python/twisted[${PYTHON_USEDEP}]
 "
 
 PATCHES=( "${FILESDIR}/${P}-libressl.patch" )
+
+python_test() {
+        "${PYTHON}" -m unittest discover || die
```
Comment (mgorny) [inline dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: `-v`, please.

File: dev-python/python-msrplib/python-msrplib-0.18.0-r1.ebuild [hunk 2]
```diff
@@ -21,6 +21,11 @@ RDEPEND="
 	dev-python/python-application[${PYTHON_USEDEP}]
 	dev-python/python-eventlib[${PYTHON_USEDEP}]
 	dev-python/python-gnutls[${PYTHON_USEDEP}]
-	dev-python/twisted-core[${PYTHON_USEDEP}]
-	dev-python/twisted-names[${PYTHON_USEDEP}]
+	dev-python/twisted[${PYTHON_USEDEP}]
 "
+
+python_test() {
+        pushd test > /dev/null
```
Comment (mgorny) [inline dev-python/python-msrplib/python-msrplib-0.18.0-r1.ebuild]: All external commands need `|| die` (i.e. those not defined by the package manager or eclasses).

File: dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild (hunk 1 above)
Comment (PikkuJose) [reply dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: I tried:

>>> Test phase: dev-python/python-msrplib-0.19.1
Usage: python -m unittest [options]

python -m unittest: error: no such option: -v
test_basic (unittest.loader.ModuleImportFailure) ... ERROR

Comment (mgorny) [reply dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: Try after `discover`.

Comment (PikkuJose) [reply dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: yes, that was clear, sorry I didn't put the code to make it easier from your side:

```
python_test() {
        pushd test > /dev/null || die
        "${EPYTHON}" -m unittest discover -v || die
        popd > /dev/null || die
}
```

Comment (mgorny) [reply dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: Let's focus on getting the missing test-dependencies first, and then I'll try to figure out how to solve the verbosity problem.

**Comment by mgorny** [comment]
Ping.

**Comment by PikkuJose** [comment]
Sorry, Michal, I revised the test and looks like I miss something or they are abandoned, so far
I didn't get them working.

**Comment by mgorny** [comment]
Please update the PR with what you've came with so far and I'll look if I can fix it some more.

**Comment by PikkuJose** [comment]
Commits updated, so far the tests looks to fail to find their own modules,
so something may be missing to allow them find the modules in the workdir.
Looks like the dependency on python2 is necessary at this stage unfortunately.

File: dev-python/python-msrplib/python-msrplib-0.19.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+DEPEND="test? ( dev-python/greenlet[${PYTHON_USEDEP}] )"
+
+python_test() {
+        pushd test > /dev/null || die
+        "${EPYTHON}" -m unittest discover -v || die
+        popd > /dev/null || die
+}
+
+src_compile() {
```
Comment (mgorny) [inline dev-python/python-msrplib/python-msrplib-0.19.1.ebuild]: Why are you doing this? If you override the whole build & install process like this, I'm not surprised it can't find anything.

File: dev-python/python-msrplib/python-msrplib-0.19.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+HOMEPAGE="http://sipsimpleclient.org"
+SRC_URI="https://github.com/AGProjects/${PN}/archive/release-${PV}.tar.gz"
+S=${WORKDIR}/${PN}-release-${PV}
+
+LICENSE="LGPL-2+"
+SLOT="0"
+IUSE="test"
+
+KEYWORDS="~amd64 ~x86"
+#RESTRICT="!test? ( test )"
```
Comment (mgorny) [inline dev-python/python-msrplib/python-msrplib-0.19.1.ebuild]: Why is this commented out?

File: dev-python/python-msrplib/Manifest [hunk 5]
```diff
@@ -1 +1 @@
-DIST python-msrplib-0.18.0.tar.gz 30234 BLAKE2B da90d00291ab68ae5e4e89f275367741d8dac21e2cb35b6b9722430147b5ada1802c943b5e224f36d565faab216c649cc6e155fc1aaf23ab2f63d3c6e39bf2a7 SHA512 ed260b0183aff543e115951f464d0e3934862439a7c049c3d19304c543805ee364129d2a99c0d132561c8ae0af9fc19cde460723f2754e973bf68a602a18db99
+DIST release-0.19.1.tar.gz 30917 BLAKE2B 8b44b5c3e0383854a0e616cc3bf6fca937fd65ca1f44a7eee8b5647db5932acffdbbcf8c1433736d932ca50e2674203b0a2ad5f2c9858b83d65818357f7015b6 SHA512 afa05606354eb82cb7d1f5c95ac1a146c25c7d22b74a44b83625351b470646e91242468f3ee0a1ccb2b3193d3a222246a92a8133614ded4747493a1f6e2b193e
```
Comment (mgorny) [inline dev-python/python-msrplib/Manifest]: Please split the removal and addition into separate commits, or preferably keep the old version for some time still.

File: dev-python/python-msrplib/python-msrplib-0.19.1.ebuild (hunk 3 above)
Comment (PikkuJose) [reply dev-python/python-msrplib/python-msrplib-0.19.1.ebuild]: it may be that time ago I felt there was need to rewrite it for any reason and it went there,
it doesn't seem to be needed anymore.

File: dev-python/python-msrplib/Manifest (hunk 5 above)
Comment (PikkuJose) [reply dev-python/python-msrplib/Manifest]: I recovered the former version on the new update

**Comment by PikkuJose** [comment]
This is right now rebased to be in sync with the master

File: dev-python/python-msrplib/python-msrplib-0.19.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+EAPI="6"
+
+PYTHON_COMPAT=( python2_7 )
+DISTUTILS_SINGLE_IMPL=1
+
+inherit distutils-r1
+
+DESCRIPTION="Client library for MSRP protocol and its relay extension"
+HOMEPAGE="http://sipsimpleclient.org"
+SRC_URI="https://github.com/AGProjects/${PN}/archive/release-${PV}.tar.gz"
```
Comment (mgorny) [inline dev-python/python-msrplib/python-msrplib-0.19.1.ebuild]: Please use `->` to rename the file to `${PN}-release-${PV}.tar.gz`. Having a distfile named `release-*` is not very readable, and can collide with other packages.

File: dev-python/python-msrplib/python-msrplib-0.19.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+S=${WORKDIR}/${PN}-release-${PV}
+
+LICENSE="LGPL-2+"
+SLOT="0"
+IUSE="test"
+
```
Comment (mgorny) [inline dev-python/python-msrplib/python-msrplib-0.19.1.ebuild]: Remove the empty line here.

File: dev-python/python-msrplib/python-msrplib-0.19.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	dev-python/python-application[${PYTHON_USEDEP}]
+	dev-python/python-eventlib[${PYTHON_USEDEP}]
+	dev-python/python-gnutls[${PYTHON_USEDEP}]
+	dev-python/twisted[${PYTHON_USEDEP}]
+"
+
+DEPEND="test? ( dev-python/greenlet[${PYTHON_USEDEP}] )"
```
Comment (mgorny) [inline dev-python/python-msrplib/python-msrplib-0.19.1.ebuild]: Are you sure that tests don't need all those RDEPEND-encies?

File: dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild [hunk 9]
```diff
@@ -1,4 +1,4 @@
-# Copyright 1999-2015 Gentoo Foundation
```
Comment (mgorny) [inline dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: Why are you calling this a 'version bump' if you're not bumping version? This is really confusing.

Comment (PikkuJose) [reply dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: how should I call a change from 0.2.1 to 0.2.1-r1?

Comment (mgorny) [reply dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: Call it by what you're actually changing.

Comment (mgorny) [reply dev-python/python-eventlib/python-eventlib-0.2.1-r1.ebuild]: I mean *meaningful* change in the ebuild. The user cares that you're changing the deps, now what revision the ebuild will have now.

**Comment by mgorny** [comment]
I'm sorry but I can't accept keeping those packages with a clear conscience. They are simply too broken, and upon running the few tests that I've been able to find you only discover how bad things are. Not to mention this hell gets deeper, with their NIH dependencies also being horribly broken. In the end, the effort put into helping you fix them is more than I can afford.

**Comment by PikkuJose** [comment]
No problems, I already got that feeling that if the upstream is in that shape that it doesn't run
the tests properly, quality is not going to be enough for serious projects like Gentoo. It can live
in my overlay meanwhile the target software still works.

On the other hand, it has been a great learning experience to me, to understand the interests
pursued when updating/creating new ebuilds. Thanks for that!

**Comment by mgorny** [comment]
Thanks. I hope to see you soon with some other packages ;-).
